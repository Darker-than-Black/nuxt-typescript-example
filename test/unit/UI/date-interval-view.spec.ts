import Vue from 'vue'
import Moment from 'moment'
import VueTestInstance from '~/test/factory/vue-instance'
import DateIntervalView from '~/components/UI/date-interval-view.vue'

describe('date-interval-view.vue', () => {
  let wrapper: any
  const dateFrom: string = '2020-07-16T13:04:33.376Z'
  beforeEach(() => {
    Vue.prototype.$moment = Moment
    wrapper = VueTestInstance(DateIntervalView, {
      options: {
        propsData: {
          dateFrom,
          dateTo: ''
        }
      }
    })
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check format date', () => {
    expect(wrapper.text()).toContain(Moment(dateFrom).format('MM.DD.YYYY h:mm'))
  })

  it('check set hyphen, when date prop empty', () => {
    expect(wrapper.text()).toContain('â€”')
  })
})
