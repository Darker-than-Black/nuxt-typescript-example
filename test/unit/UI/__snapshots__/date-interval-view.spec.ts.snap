// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date-interval-view.vue should match the snapshot 1`] = `
<div class="d-flex align-items-center"><svg width="10" height="26" viewBox="0 0 10 26" fill="none" xmlns="http://www.w3.org/2000/svg" class="d-inline-block mr-2 ">
    <path d="M7 13C7 10.925 8.49188 9.13589 9.44461 7.29257C9.79955 6.60585 10 5.82633 10 5C10 2.23858 7.76142 0 5 0C2.23858 0 0 2.23858 0 5C0 5.82633 0.200454 6.60585 0.555388 7.29257C1.50812 9.13589 3 10.925 3 13C3 15.075 1.50812 16.8641 0.555389 18.7074C0.200454 19.3942 0 20.1737 0 21C0 23.7614 2.23858 26 5 26C7.76142 26 10 23.7614 10 21C10 20.1737 9.79955 19.3942 9.44461 18.7074C8.49188 16.8641 7 15.075 7 13Z" fill="white" class="svg-path"></path>
    <circle cx="5" cy="5" r="3" fill="#5FCB09"></circle>
    <circle cx="5" cy="21" r="3" fill="#CCC"></circle>
  </svg>
  <div class="d-inline-block">
    07.16.2020 4:04
    <br>
    â€”
  </div>
</div>
`;
